airbyte-integrations/postgres-destination/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java

```diff
@@ -0,0 +1,343 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.postgres;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Charsets;
+import io.airbyte.commons.concurrency.GracefulShutdownHandler;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.config.DestinationConnectionSpecification;
+import io.airbyte.config.Schema;
+import io.airbyte.config.StandardCheckConnectionOutput;
+import io.airbyte.config.StandardCheckConnectionOutput.Status;
+import io.airbyte.config.StandardDiscoverSchemaOutput;
+import io.airbyte.config.Stream;
+import io.airbyte.db.DatabaseHelper;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.DestinationConsumer;
+import io.airbyte.integrations.base.FailureTrackingConsumer;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.persistentqueue.BigQueue;
+import io.airbyte.persistentqueue.CloseableInputQueue;
+import io.airbyte.singer.SingerMessage;
+import io.airbyte.singer.SingerMessage.Type;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PostgresDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PostgresDestination.class);
+  static final String COLUMN_NAME = "data";
+
+  @Override
+  public DestinationConnectionSpecification spec() throws IOException {
+    // return a jsonschema representation of the spec for the integration.
+    final String resourceString = MoreResources.readResource("spec.json");
+    return Jsons.deserialize(resourceString, DestinationConnectionSpecification.class);
+  }
+
+  @Override
+  public StandardCheckConnectionOutput check(JsonNode config) {
+    try {
+      final BasicDataSource connectionPool = getConnectionPool(config);
+      DatabaseHelper.query(connectionPool, ctx -> ctx.execute(
+          "SELECT *\n"
+              + "FROM pg_catalog.pg_tables\n"
+              + "WHERE schemaname != 'pg_catalog' AND schemaname != 'information_schema';"));
+
+      connectionPool.close();
+    } catch (Exception e) {
+      // todo (cgardens) - better error messaging for common cases. e.g. wrong password.
+      return new StandardCheckConnectionOutput().withStatus(Status.FAILURE).withMessage(e.getMessage());
+    }
+
+    return new StandardCheckConnectionOutput().withStatus(Status.SUCCESS);
```

Comment:
Can we move this into the end of the `try` so the method reads linearly except in the exceptional case?